### DISCUSSION TITLE
Recording V3.8 latency / quality Closed

### DISCUSSION THREAD

**Discussion by reinhardwh** [jamulussoftware/jamulus#1883] [Bug Investigation]
Hello,
Upgraded Server from 3.7 to 3.8
Since then we noticed issues during recordings of sessions.

- high latency
- sound quality deteriorates

When stopping recordings quality goes up immediately

Thanks for looking into this

**Comment by ghost** [comment]
Interesting report. Which operating system are you using?

**Reply by reinhardwh** [reply]
Name in Rock Genre is 
SoR-Neussy
Linux 3.8
178.254.13.132

**Reply by ghost** [reply]
Try: Jamulus --help
to see the command line options.

Maybe: Jamulus -T
can help with performance.

Maybe: Jamulus -P
is making a high cpu load.

**Reply by reinhardwh** [reply]
Did not change Server parameters when updating..

I use
- T Multithreading **enabled**
- F Fast update to support small network package **enabled**

I do not use -P

**Reply by dcorson-ticino-com** [reply]
I would use top to take a look at your processor use.
That sounds a lot like too many background tasks running.

**Reply by reinhardwh** [reply]
Checked processor load using htop
4 cores show max 30% load each
4-6 musicians connected

Its a root server - only running jamulus

Starting the recording does not have an affect on CPU load
But its immediately audible when recording is started - for all musicians

**Reply by dcorson-ticino-com** [reply]
Is that running on a old Pi ? otherwise that is a very high processor load for so few musicians.

**Reply by reinhardwh** [reply]
Its a root server KVM

Recording worked fine on version 3.7

Problem started after the upgrade to 3.8

**Comment by chrisrimple** [comment]
FWIW, I destroyed and recreated my Debian VM (on Vultr.com), then installed 3.8.0 with recording always on.  During last night's session with 6 musicians, I see that CPU load increased to 90%.  Can't say if that was true on earlier versions, but that does seem high.

**Reply by reinhardwh** [reply]
Did you notice disturbances or bad sound quality like me?

**Reply by chrisrimple** [reply]
Occasionally, but that happened at times on earlier versions as well, so I attributed it to internet hiccups.  We rehearse 2x/week, and I'll watch CPU load live during our next session.

**Reply by hoffie** [reply]
Any updates here?
It would be interesting to know
- If downgrading fixes it
- If the issue is reproducible (run with recording for X hours, observe high CPU usage)

The recorder part has been touched as part of 3.8.0, so it is certainly possible that some bug has slipped through, although I cannot think of anything obvious.

If overall resource usage is not an issue on that machine, my first suspicion would be some synchronization issue -- e.g. main Jamulus waiting for the recorder thread to finish work (which does lots of IO which may be blocking, obviously).

**Reply by chrisrimple** [reply]
I've had a few 1+ hour sessions with 6 musicians on 3.8.0 now, all with recording enabled, and CPU never exceeded 30%.  I've heard the occasional hiccup, but I attribute that to network/jitter, not CPU load.

**Reply by reinhardwh** [reply]
Hello,
thanks for your comments and looking into this.
Today I found:
Most likely caused by my server configuration.
Started 2 Jamulus instances after reboot...
Shame on me...
Case closed.
